bluemira/equilibria/optimisation/constraints.py

```diff
@@ -445,6 +447,69 @@ def f_constraint(self) -> ConstraintFunction:
         return f_constraint
 
 
+class SphericalHarmonicsConstraint(MagneticConstraint):
+    """
+    Constraint function to constrain spherical harmonics starting from initial
+    coil currents and associated core plasma.
+
+    Parameters
+    ----------
+    r_t:
+        Typical length scale of the problem (e.g. radius at outer midplane)
+    ref_harmonics:
+        Initial harmonic amplitudes obtained from desired core plasma
+    max_degree:
+        Maximum degree of spherical harmonics desired to constrain.
+    """
+
+    def __init__(
+        self,
+        r_t: float,
+        ref_harmonics: npt.NDArray,
+        max_degree: int,
+        target_value: float = 0.0,
+        weights: Union[float, npt.NDArray] = 1.0,
+        tolerance: Union[float, npt.NDArray] = 1e-6,
+        f_constraint: Type[ConstraintFunction] = AxBConstraint,
+        constraint_type: str = "inequality",
+    ):
+        super().__init__(
+            target_value,
+            weights,
+            tolerance=tolerance,
+            f_constraint=f_constraint,
+            constraint_type=constraint_type,
+        )
+        self.max_degree = max_degree
+        self.r_t = r_t
+        self.ref_harmonics = ref_harmonics
```

Comment:
After to talking with Georgi, I think "harmonics" is a slightly misleading label. They are more amplitudes, or perhaps `ref_coil_psi_amplitudes``

As they are the SH amplitudes of psi w.r.t the current in the coils